pkg/await/await.go

```diff
@@ -299,28 +307,39 @@ func Deletion(
 		version = client.DefaultVersion()
 	}
 
+	// Manually set delete propagation for Kubernetes versions < 1.6 to avoid bugs.
 	deleteOpts := metav1.DeleteOptions{}
 	if version.Compare(1, 6) < 0 {
 		// 1.5.x option.
 		boolFalse := false
 		// nolint
 		deleteOpts.OrphanDependents = &boolFalse
-	} else {
+	} else if version.Compare(1, 7) < 0 {
 		// 1.6.x option. (NOTE: Background delete propagation is broken in k8s v1.6, and maybe later.)
 		fg := metav1.DeletePropagationForeground
 		deleteOpts.PropagationPolicy = &fg
 	}
 
 	// Obtain client for the resource being deleted.
 	clientForResource, err := client.FromGVK(pool, disco, gvk, namespace)
-	if err != nil {
+	if errors.IsNotFound(err) {
```

Comment:
@lblackstone Did you mean:

```
	if err != nil {
		if errors.IsNotFound(err) {
			// If the resource G/V/K is not found, but was successfully created before, (e.g., it
			// was a CR, but the CRD has since been removed) we can assume that it has now been
			// successfully deleted.
			return nil
		}
		return err
	}
```

Reply:
Oh, I think you meant this snippet to be for the similar code block below.

Reply:
Oops, you're right.